### PR TITLE
Parallelize receiving secure octo streams

### PR DESCRIPTION
**Pull request by JonathanLennox**

### CODE REVIEW

File: jvb/src/main/kotlin/org/jitsi/videobridge/relay/Relay.kt [hunk 1]
```diff
@@ -337,6 +341,34 @@ class Relay @JvmOverloads constructor(
...
+     */
+    private fun handleMediaPacket(packetInfo: OctoPacketInfo) {
+        if (packetInfo.packet.looksLikeRtp()) {
+            val ssrc = RtpHeader.getSsrc(packetInfo.packet.buffer, packetInfo.packet.offset)
+            val ep = getEndpointBySsrc(ssrc)
+            if (ep != null) {
+                ep.handleIncomingPacket(packetInfo)
+                return
+            } else {
+                logger.warn { "RTP Packet received for unknown endpoint SSRC $ssrc" }
```
Comment (bgrozev) [inline jvb/src/main/kotlin/org/jitsi/videobridge/relay/Relay.kt:355]: Return the buffer?

File: jvb/src/main/kotlin/org/jitsi/videobridge/relay/Relay.kt [hunk 2]
```diff
@@ -453,14 +536,27 @@ class Relay @JvmOverloads constructor(
...
 
-    /** TODO this is inefficient, will be better when we have per-endpoint transceivers. */
-    private fun updateTransceiverSources() {
-        val mediaSources = ArrayList<MediaSourceDesc>()
-        relayedEndpoints.values.forEach { r -> mediaSources.addAll(r.mediaSources) }
-        transceiver.setMediaSources(mediaSources.toTypedArray())
+    fun addPayloadType(payloadType: PayloadType) {
+        transceiver.addPayloadType(payloadType)
+        payloadTypes.add(payloadType)
+        relayedEndpoints.values.forEach { ep -> ep.addPayloadType(payloadType) }
```
Comment (bgrozev) [inline jvb/src/main/kotlin/org/jitsi/videobridge/relay/Relay.kt:544]: Should this aquire `endpointsLock` (same for addRtpExtension below)?

File: jvb/src/main/kotlin/org/jitsi/videobridge/relay/Relay.kt [hunk 3]
```diff
@@ -584,6 +687,22 @@ class Relay @JvmOverloads constructor(
         val queueErrorCounter = CountingErrorHandler()
     }
 
+    class Statistics {
+        val bytesReceived = AtomicLong(0)
```
Comment (bgrozev) [inline jvb/src/main/kotlin/org/jitsi/videobridge/relay/Relay.kt:691]: These are only updated on expire?

Comment (JonathanLennox) [reply jvb/src/main/kotlin/org/jitsi/videobridge/relay/Relay.kt:691]: Yeah.

**Review by bgrozev — APPROVED** [review]

**Pull request merged**